script/tool/lib/src/update_excerpts_command.dart

```diff
@@ -23,232 +26,302 @@ class UpdateExcerptsCommand extends PackageLoopingCommand {
+                  '${file.path}:$lineNumber: code block has wrong language');
+            }
+            mode = _ExcerptParseMode.injecting;
+          }
+          output.writeln(line);
+          break;
+        case _ExcerptParseMode.injecting:
+          if (line == '```') {
+            if (existingBlock.toString() != excerpt) {
+              detectedChange = true;
+            }
+            output.write(excerpt);
+            output.writeln(line);
+            mode = _ExcerptParseMode.normal;
+            language = null;
+            excerpt = null;
+            existingBlock.clear();
+          } else {
+            existingBlock.writeln(line);
+          }
+          break;
+      }
     }
-    final Set<String> submoduleDependencies = <String>{
-      'code_excerpter',
-      'code_excerpt_updater',
-    };
-    final String relativeRootPath =
-        getRelativePosixPath(repoRoot, from: package.directory);
-    for (final String dependency in submoduleDependencies) {
-      editablePubspec.update(<String>[
-        devDependenciesKey,
-        dependency
-      ], <String, String>{
-        'path': '$relativeRootPath/site-shared/packages/$dependency'
-      });
+    if (detectedChange) {
+      if (errors != null) {
+        reportError('${file.path}: skipped updating file due to errors');
+      } else {
+        try {
+          file.writeAsStringSync(output.toString());
+        } catch (e) {
+          reportError(
+              '${file.path}: failed to update file (${e.runtimeType}: $e)');
+        }
+      }
     }
-    package.pubspecFile.writeAsStringSync(editablePubspec.toString());
-  }
-
-  /// Restores the version of the pubspec that was present before running
-  /// [_addSubmoduleDependencies].
-  void _undoPubspecChanges(RepositoryPackage package) {
-    package.directory
-        .childFile(_originalPubspecFilename)
-        .renameSync(package.pubspecFile.path);
+    return _UpdateResult(detectedChange, errors);
   }
 
-  /// Checks the git state, returning an error string if any .md files have
-  /// changed.
-  Future<String?> _validateRepositoryState(RepositoryPackage package) async {
-    final io.ProcessResult checkFiles = await processRunner.run(
-      'git',
-      <String>['ls-files', '--modified'],
-      workingDir: package.directory,
-      logOnError: true,
-    );
-    if (checkFiles.exitCode != 0) {
-      return 'Unable to determine local file state';
+  String _extractExcerpt(
+      File excerptSourceFile,
+      String section,
+      String plasterInside,
+      String language,
+      void Function(String) reportError) {
+    final List<String> buffer = <String>[];
+    bool extracting = false;
+    int lineNumber = 0;
+    int maxLength = 0;
+    bool found = false;
+    String prefix = '';
+    String suffix = '';
+    String padding = '';
+    switch (language) {
+      case 'c++':
+      case 'dart':
+      case 'js':
+      case 'kotlin':
+      case 'rfwtxt':
+      case 'scss':
```

Comment:
What is this? We don't have any in the repo.

Reply:
dunno, this list was just the list from the original package plus `rfwtxt` (and now `cc`)